scripts/tests/create_project_table_test.py

```diff
@@ -0,0 +1,139 @@
+"""
+Unit tests create_project_table.py.
+
+This suite tests functions for extracting metadata, generating markdown table rows, 
+processing multiple README files, and inserting rows into tables, using mocks to simulate file operations.
+"""
+
+import unittest
+from unittest.mock import MagicMock, patch
+from pathlib import Path
+import sys
+
+sys.path.append(str(Path(__file__).resolve().parent.parent))
+
+from create_project_table import (
+    extract_metadata,
+    to_table_row,
+    generate_readme_table,
+    insert_rows_to_table,
+)
+
+
+class TestCreateProjectTable(unittest.TestCase):
+    def test_extract_metadata(self):
+        mock_readme_content = """\
+title: Example Project
+description: An example project for testing.
+integrations: "slack", "gemini"
+extra_key: This should not appear
+"""
+
+        mock_readme_file = MagicMock()
+        mock_readme_file.read_text.return_value = mock_readme_content
+
+        expected_metadata = {
+            "title": "Example Project",
+            "description": "An example project for testing.",
+            "integrations": ["slack", "gemini"],
+        }
+
+        metadata = extract_metadata(mock_readme_file)
+        self.assertEqual(metadata, expected_metadata)
+
+    def test_to_table_row(self):
+        mock_readme_file = Path("./projects/project1/README.md")
+
+        metadata = {
+            "title": "Project One",
+            "description": "An example project for testing.",
+            "integrations": ["slack", "gemini"],
+        }
+
+        expected_row = "| [Project One](./projects/project1/) | An example project for testing. | slack, gemini |\n"
+        row = to_table_row(mock_readme_file, metadata)
+        self.assertEqual(row, expected_row)
+
+    @patch("create_project_table.Path.rglob")
+    @patch("create_project_table.extract_metadata")
+    @patch("create_project_table.to_table_row")
```

Comment:
Mocks are a powerful testing aid, but they're unnecessary and make testing harder and limited in this case. Let's discuss alternatives offline.